test-suite/djinni/foo_duplicate_file_creation.djinni

```diff
@@ -0,0 +1,13 @@
+Foo_Record = record {
```

Comment:
The names in this file should generally be `snake_case` to allow Djinni to do its standard style-selection for file-names and identifiers.  That's assumed when generating names, though not enforced.  I suspect that's why the bug you encountered was never seen before, since we hadn't tested your use case.

Given this test file is trying to verify the fix for this bug, I think it's right to keep its names mixed-case.  However I'd suggest adding a doc comment describing why this is the case, for future reference.  Otherwise I wouldn't be surprised if someone starts to enforce the name requirement, and comes across this test and gets confused by why it's here.

While you're at it, and explanation about the duplicate list types here, and what they're testing would be welcome.

Reply:
Done